src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -272,6 +269,102 @@ export class CollapseCellsAction extends ToggleableAction {
 	}
 }
 
+// Run Notebook with Parameters
+export class RunParametersAction extends TooltipFromLabelAction {
+	private static readonly label = localize('runParameters', "Run with Parameters");
+	private static readonly iconClass = 'icon-run-with-parameters';
+
+	constructor(id: string, toggleTooltip: boolean,
+		context: URI,
+		@IQuickInputService private quickInputService: IQuickInputService,
+		@INotebookService private _notebookService: INotebookService,
+		@INotificationService private notificationService: INotificationService,
+	) {
+		super(id, {
+			label: RunParametersAction.label,
+			baseClass: baseIconClass,
+			iconClass: RunParametersAction.iconClass,
+			maskedIconClass: maskedIconClass,
+			shouldToggleTooltip: toggleTooltip
+		});
+	}
+
+	/**
+	 * Gets Default Parameters in Notebook from Parameter Cell
+	 * Uses that as Placeholder values for user to inject new values for
+	 * Once user enters all values it will open the new parameterized notebook
+	 * with injected parameters value from the QuickInput
+	*/
+	public async run(context: URI): Promise<void> {
+		// Get Parameters via Notebook Model
+		const editor = this._notebookService.findNotebookEditor(context);
+		// Set Default Parameters to the mapped string
+		let defaultParameters = new Map<string, string>();
+		editor.cells.forEach(cell => {
+			if (cell.isParameter) {
+				for (let parameter of cell.source) {
+					let param = parameter.trim().split('=');
```

Comment:
I don't think you need this first `trim()`

Reply:
Also, I'm not sure if we want to take into account string values that could have the equal sign? 
Maybe splitting the string is not the best option.

Reply:
The first trim is needed as the parameters will come in as ['x = 1 \n', 'y = 2'], as a result the first parameter when we parse will make the string have a new line. I think ensuring that both are trimmed ensures that there is no additional space added when adding to the map. 

Also, I think the scenario where values have equal sign would be an edge case but I could focus on splitting on the first instance of the = sign instead.